baseplate/sidecars/live_data_watcher.py

```diff
@@ -128,25 +73,67 @@ def on_change(self, data: Optional[bytes], _znode_stat: ZnodeStat) -> None:
             os.fchmod(tmpfile.fileno(), self.mode)
-            try:
-                json_data = json.loads(data.decode("UTF-8"))
-            except json.decoder.JSONDecodeError:
-                # If JSON fails to decode, still write the bytes data since
-                # we don't necessarily know if the the contents of the znode
-                # had to be valid JSON anyways.
-                tmpfile.write(data)
-            else:
-                # If no exceptions, we have valid JSON, and can parse accordingly.
-                data = NodeWatcher.get_data_to_write(json_data, data)
-                if data is None:
-                    logger.warning(
-                        "No data written to destination file. Something is likely misconfigured."
-                    )
-                    return
-                tmpfile.write(data)
+            tmpfile.write(data)
         os.rename(self.dest + ".tmp", self.dest)
 
 
+def _parse_loader_type(data: Optional[bytes]) -> LoaderType:
+    try:
+        json_data = json.loads(data.decode("UTF-8"))
+    except json.decoder.JSONDecodeError:
+        logger.debug("Data is not valid JSON. Loading as RAW")
+        return LoaderType.RAW
+
+    try:
+        loader_type = json_data["live_data_watcher_load_type"]
+    except (json.decoder.JSONDecodeError, KeyError, AttributeError, TypeError):
+        logger.debug("Data is not a JSON object like {\"live_data_watcher_load_type\":\"load_type\"}. Loading as RAW")
+        return LoaderType.RAW
+
+    return LoaderType(loader_type)
+
+
+def _load_from_s3(data: bytes) -> Optional[bytes]:
+    loader_config = json.loads(data.decode("UTF-8"))
+    try:
+        region_name = loader_config["region_name"]
+        s3_kwargs = {
+            "Bucket": loader_config["bucket_name"],
+            "Key": loader_config["file_key"],
+            "SSECustomerKey": loader_config["sse_key"],
+            "SSECustomerAlgorithm": "AES256",
+        }
+    except KeyError as e:
+        # We require all of these keys to properly read from S3.
+        logger.exception(
+                "Failed to update live config: unable to fetch content from s3: source config has invalid or missing keys: %s.",
+                e.args[0]
+        )
+        return None
+
+    s3_client = boto3.client(
+        "s3",
+        region_name=region_name,
+    )
+
+    try:
+        s3_object = s3_client.get_object(**s3_kwargs)
+        # Returns bytes.
+        return s3_object["Body"].read()
+    except ClientError as error:
+        logger.exception(
+            "Failed to update live config: failed to load data from S3. Received error code %s: %s",
+            error.response["Error"]["Code"],
+            error.response["Error"]["Message"],
+        )
+    except ValueError as error:
+        logger.exception(
+            "Failed to update live config: params for loading from S3 are incorrect. Received error: %s",
+            error,
+        )
+    return None
```

Comment:
I would argue that if it failed to load data, it should return an exception, not None, which is a valid piece of data.